diagnostic.js

```diff
@@ -29,30 +29,79 @@ mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
+const Movement = require('./models/movement.js')
+
+const done = function() {
+  db.close();
+};
 
 const create = (name, description, startYear, endYear) => {};
 // Success -> Print new Movement as JSON
 // Failure -> Console.error
+  Movement.create({
+    name: name,
+    description: description,
+    startYear: startYear,
+    endYear: endYear.
+  }).then(function(err, movement) {
+    console.log(movement.toJSON());
+  }).catch(console.error).then(done);
 
-const index = () => {};
+const index = () => {
 // Success -> Print all Movements as JSON
 // Failure -> Console.error
+  Movement.find().then(function(movement) {
+    movement.forEach(function(movement) {
+      console.log(movement.toJSON());
+    });
+  }).catch(console.error).then(done);
+};
+
 
-const show = (id) => {};
+const show = (id) => {
 // Success -> If the specified Movement exists, print it as JSON;
 //              otherwise, print "Not Found" and exit.
 // Failure -> Console.error
+  if (Movement.findById(id)) {
+    Movement.findById(id).then(function() {
+     console.log(movement.toJSON());
+   }).catch(console.error).then(done);
+ }
+ else {
+   console.log("Not Found");
+ }
```

Comment:
You've got the right idea, but the implementation is wrong. 
`Movement.findById(id)` returns a query. And even if you added `.exec()`, you'd get a Promise as the result even if it went unfulfilled. You'll need to test for whether or not a match was found in the callback.
```
Movement.findById(id).then(function(movement) {
  if (!movement) { console.log("NOT FOUND"); return; }
  console.log(movement.toJSON());
 }).catch(console.error).then(done);
```